docs/articles/documentation/guides/basic-guides/assert.md

```diff
@@ -0,0 +1,145 @@
+
+This topic consists of the following sections:
+
+* [Assertion Structure](#assertion-structure)
+* [Smart Assertion Query Mechanism](#smart-assertion-query-mechanism)
+
+## Assertion Structure
+
+To construct assertions, use the test controller's [`expect`](../../reference/test-api/testcontroller/expect/README.md) method.
+
+The `expect` method is followed by an assertion method that accepts an expected value
+and optional arguments.
+
+The following example shows the structure of the deep equality assertion:
+
+```text
+await t.expect( actual ).eql( expected, message, options );
+```
+
+The code snippet below demonstrates how to use assertions in a test:
+
+```js
+import { Selector } from 'testcafe';
+
+
+fixture `Example page`
+   .page `http://devexpress.github.io/testcafe/example/`;
+
+
+test('Check property of element', async t => {
+   const developerNameInput = Selector('#developer-name');
+
+   await t
+       .expect(developerNameInput.value).eql('', 'input is empty')
+       .typeText(developerNameInput, 'Peter Parker')
+       .expect(developerNameInput.value).contains('Peter', 'input contains text "Peter"');
+});
+```
+
+The following assertion methods are available:
+
+* [Deep Equal](../../reference/test-api/testcontroller/expect/eql.md)
+* [Not Deep Equal](../../reference/test-api/testcontroller/expect/noteql.md)
+* [Ok](../../reference/test-api/testcontroller/expect/ok.md)
+* [Not Ok](../../reference/test-api/testcontroller/expect/notok.md)
+* [Contains](../../reference/test-api/testcontroller/expect/contains.md)
+* [Not Contains](../../reference/test-api/testcontroller/expect/notcontains.md)
+* [Type of](../../reference/test-api/testcontroller/expect/typeof.md)
+* [Not Type of](../../reference/test-api/testcontroller/expect/nottypeof.md)
+* [Greater than](../../reference/test-api/testcontroller/expect/gt.md)
+* [Greater than or Equal to](../../reference/test-api/testcontroller/expect/gte.md)
+* [Less than](../../reference/test-api/testcontroller/expect/lt.md)
+* [Less than or Equal to](../../reference/test-api/testcontroller/expect/lte.md)
+* [Within](../../reference/test-api/testcontroller/expect/within.md)
+* [Not Within](../../reference/test-api/testcontroller/expect/notwithin.md)
+* [Match](../../reference/test-api/testcontroller/expect/match.md)
+* [Not Match](../../reference/test-api/testcontroller/expect/notmatch.md)
+
+## Smart Assertion Query Mechanism
+
+In synchronous functional testing, you can execute assertions immediately after test actions.
+
+![Synchronous Functional Testing](../../../images/assertions/synchronous-testing.png)
+
+Functional tests are asynchronous on the web. This means that we cannot get the expected changes immediately after an end user's actions.
+For example, it can take time for the tested page to send a request to the server for the required data, or an end user's action launches an animation after which the page reaches its final state.
```

Comment:
```suggestion
For example, it can take time for the tested page to send a request to the server for the required data or a user's action to invoke an animation after which the page reaches its final state.
```